building/build-debs/homeworld-admin-tools/src/configuration.py

```diff
@@ -164,6 +164,27 @@ def get_node(self, node_name: str) -> Node:
                 return node
         command.fail("no such node: %s" % node_name)
 
+    def has_hostname(self, hostname: str) -> bool:
+        if hostname.endswith(self.external_domain):
+            # strip external domain
+            hostname = hostname[:-(len(self.external_domain) + 1)]
+        elif hostname.endswith(self.internal_domain):
+            # strip internal domain
+            hostname = hostname[:-(len(self.internal_domain) + 1)]
+        return self.has_node(hostname)
+
+    def get_hostname(self, name: str) -> str:
+        hostname = name
+        if name.endswith(self.external_domain):
+            # strip external domain
+            hostname = name[:-(len(self.external_domain) + 1)]
```

Comment:
You're stripping out the dot in the domain name, but not checking that it exists. Given an `external_domain` of `mit.edu`, this could lead to a situation where something like `test.commit.edu` gets converted into `test.co`, which isn't good behavior.

Reply:
oops good catch